### DISCUSSION TITLE
U8G2_R0 and I2C address

### DISCUSSION THREAD

**Discussion by jandemna** [olikraus/u8g2#1588] [Q&A] [answered]
My .96" OLED display (2 euro:) worked directly with defined line below attached to D1 MINI.  Thanks olikraus for the nice lib!! 

U8G2_SSD1306_128X64_NONAME_F_SW_I2C u8g2(U8G2_R0, /* clock=*/ SCL, /* data=*/ SDA, /* reset=*/ U8X8_PIN_NONE);   // All Boards without Reset of the Display

What is U8G2_R0 and where is it defined? Looked for it everywhere, but not successful. Does it define the I2C address or is this fixed in SSD 1306?

**Comment by olikraus** [comment, marked as answer]
U8G2_R0 (and friends) are described here: https://github.com/olikraus/u8g2/wiki/u8g2reference#carduino-example

Declarations are here: https://github.com/olikraus/u8g2/blob/44860472e14cb3f7067add02fcc4ae941c6bddcb/csrc/u8g2.h#L435-L447

Usually the lowest available I2C address is used as default in U8g2. Any other I2C address can be assigned with https://github.com/olikraus/u8g2/wiki/u8g2reference#seti2caddress

**Reply by jandemna** [reply]
Thanks, very clear. I Need to improve on my search (tools).....and RTFM
